corehq/apps/users/tests/test_role_utils.py

```diff
@@ -118,3 +125,75 @@ def _delete_presets(self):
         for role in UserRole.objects.get_by_domain(self.domain):
             if role.id != self.role1.id:
                 role.delete()
+
+
+class TestCommcareAnalyticsRolesByUser(TestCase):
+
+    DOMAIN = "domain1"
+    USERNAME = "username"
+    PASSWORD = "***"
+
+    @classmethod
+    def setUpClass(cls):
+        cls.domain = create_domain(cls.DOMAIN)
+        cls.user = WebUser.create("domain1", cls.USERNAME, cls.PASSWORD, None, None)
+
+        cls.hq_no_cca_role = cls._create_role("No CCA role", analytics_roles=None)
+
+        cls.limited_cca_roles = [
+            COMMCARE_ANALYTICS_GAMMA,
+            COMMCARE_ANALYTICS_SQL_LAB,
+        ]
+        cls.hq_limited_cca_role = cls._create_role("Limited CCA role", analytics_roles=cls.limited_cca_roles)
+        cls.hq_all_cca_roles_role = cls._create_role("All CCA roles", analytics_roles=[], all_roles=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        for role in UserRole.objects.get_by_domain(cls.domain):
+            role.delete()
+
+        cls.user.delete(deleted_by_domain=cls.domain.name, deleted_by=None)
+        cls.domain.delete()
```

Comment:
`tearDownClass()` needs to call the superclass's `tearDownClass()` method to roll back the test's database transaction (among other things).

```suggestion
        cls.domain.delete()
        super().tearDownClass()
```